GitHubDeployments.ps1

```diff
@@ -0,0 +1,567 @@
+        You can list up to six users or teams as reviewers.
+        The reviewers must have at least read access to the repository.
+        Only one of the required reviewers needs to approve the job for it to proceed.
+
+    .PARAMETER ReviewerUserId
+        The users that may review jobs that reference the environment.
+        You can list up to six users or teams as reviewers.
+        The reviewers must have at least read access to the repository.
+        Only one of the required reviewers needs to approve the job for it to proceed.
+
+    .PARAMETER PassThru
+        Returns the updated environment.  By default, the Set-GitHubDeploymentEnvironment cmdlet
+        does not generate any output.
+        You can use "Set-GitHubConfiguration -DefaultPassThru" to control the default behavior
+        of this switch.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .INPUTS
+        GitHub.Branch
+        GitHub.Content
+        GitHub.Event
+        GitHub.Issue
+        GitHub.IssueComment
+        GitHub.Label
+        GitHub.Milestone
+        GitHub.PullRequest
+        GitHub.Project
+        GitHub.ProjectCard
+        GitHub.ProjectColumn
+        GitHub.Reaction
+        GitHub.Release
+        GitHub.ReleaseAsset
+        GitHub.Repository
+
+    .OUTPUTS
+        GitHub.DeploymentEnvironment
+
+    .EXAMPLE
+        New-GitHubDeploymentEnvironment -OwnerName microsoft -RepositoryName PowerShellForGitHub -EnvironmentName 'Test'
+
+        Creates a new deployment environment called 'Test' for the specified repo.
+#>
+
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParameterSetName = 'Elements')]
+    [OutputType({ $script:GitHubDeploymentEnvironmentTypeName })]
+    [Alias('Set-GitHubDeploymentEnvironment')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "", Justification="The Uri parameter is only referenced by Resolve-RepositoryElements which get access to it from the stack via Get-Variable -Scope 1.")]
+    param(
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'Uri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline,
+            ValueFromPipelineByPropertyName)]
+        [ValidateNotNullOrEmpty()]
+        [string] $EnvironmentName,
+
+        [ValidateRange(0, 43200)]
+        [int32] $WaitTimer,
+
+        [ValidateSet('ProtectedBranches', 'CustomBranchPolicies', 'None')]
+        [string] $DeploymentBranchPolicy,
+
+        [array] $ReviewerTeamId,
```

Comment:
Is there a situation where the module is providing the user back an object where the Team or User ID is a string?  (Legitimate question here).  If not, it might make more sense to be explicit with the type expectations here as `[int32[]]`.